### PR TITLE
Bug 1834300 - Update Search Engine URLs for Daum, Gule sider and Vatera.hu.

### PR DESCRIPTION
**Pull request by Standard8**
Updates some application provided search engine URLs. These don't have tests as it is not possible to test external dependencies easily, and we only have stricter checks for the main engines.

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [X] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [X] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [X] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1834300

### CODE REVIEW

File: android-components/components/feature/search/src/main/assets/searchplugins/daum-kr.xml [hunk 1]
```diff
@@ -3,7 +3,7 @@
    - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
 
 <SearchPlugin xmlns="http://www.mozilla.org/2006/browser/search/">
-<ShortName>다음지도</ShortName>
+<ShortName>다음</ShortName>
```
Comment (Standard8) [inline android-components/components/feature/search/src/main/assets/searchplugins/daum-kr.xml:6]: The name changed here to reflect what the search engine is not actually searching a map (to reflect what Desktop does).

**Comment by sarah541** [comment]
Compared this against Desktop and LGTM!

File: android-components/components/feature/search/src/main/assets/searchplugins/vatera.xml [hunk 2]
```diff
@@ -8,7 +8,9 @@
...
-  <Url type="text/html" method="GET" template="http://m.vatera.hu/page/advanced_search/action/advancedSearch/">
+  <Url type="text/html" method="GET" template="https://www.vatera.hu/listings/index.php">
+    <Param name="c" value="0" />
+    <Param name="td" value="on" />
     <Param name="q" value="{searchTerms}" />
```
[the remark this answers is not in the document]
Comment (Standard8) [reply android-components/components/feature/search/src/main/assets/searchplugins/vatera.xml:14]: At some stage we should update desktop as well.

The order of parameters shouldn't generally matter, but we've found seen previously (especially on desktop), that sometimes people appear to edit the end of URLs (presumably to search). Hence having the search term on the end works slightly better in that situation.

[the remark this answers is not in the document]
Comment (sarah541) [reply android-components/components/feature/search/src/main/assets/searchplugins/vatera.xml:14]: I concord! Thank you  for the explanation.

**Comment by Standard8** [comment]
@mergifyio backport releases_v114

**Comment by rvandermeulen** [comment]
@mergifyio backport releases_v114

**Pull request merged**